lib/modules/datasource/hermit/index.ts

```diff
@@ -0,0 +1,144 @@
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { GithubHttp } from '../../../util/http/github';
+import { regEx } from '../../../util/regex';
+import { streamToString } from '../../../util/streams';
+import { id } from '../../versioning/hermit';
+import { Datasource } from '../datasource';
+import { GithubReleasesDatasource } from '../github-releases';
+import { getApiBaseUrl } from '../github-releases/common';
+import type {
+  GithubRelease,
+  GithubReleaseAsset,
+} from '../github-releases/types';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { HermitSearchResult } from './types';
+
+/**
+ * Hermit Datasource searches a given package from the specified `hermit-pakcages`
+ * repository. It expects the search manifest to come from an asset `index.json` from
+ * a release named index. Any error fetching, parsing and finding the package from the given
+ * registryUrl will be thrown
+ */
+export class HermitDatasource extends Datasource {
+  static readonly id = 'hermit';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = id;
+
+  override readonly defaultRegistryUrls = [
+    'https://github.com/cashapp/hermit-packages',
+  ];
+
+  pathRegex: RegExp;
+
+  constructor() {
+    super(HermitDatasource.id);
+    this.http = new GithubHttp(GithubReleasesDatasource.id);
+    this.pathRegex = regEx('^\\/(?<owner>[^/]+)\\/(?<repo>[^/]+)$');
+  }
+
+  @cache({
+    namespace: `datasource-hermit-package`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? ''}-${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    logger.trace(`HermitDataSource.getReleases()`);
+
+    if (!registryUrl) {
+      logger.error('registryUrl must be supplied');
+      return null;
+    }
+
+    if (!registryUrl.startsWith('https://github.com/')) {
+      logger.error({ registryUrl }, 'Only Github registryUrl is supported');
+      return null;
+    }
+
+    const items = await this.getHermitSearchManifest(registryUrl);
+
+    if (items === null) {
+      return null;
+    }
+
+    const res = items.find((i) => i.Name === packageName);
+
+    if (!res) {
+      logger.error({ packageName, registryUrl }, 'cannot find hermit package');
```

Comment:
```suggestion
      logger.debug({ packageName, registryUrl }, 'cannot find hermit package');
```
this isn't an error. we usually don't fail here. logging an error would cause renovate to exit with error.